docs/agent-vs-aea.md

```diff
@@ -199,7 +200,8 @@ def run():
     )
 
     # Create our Agent
-    my_agent = MyAgent("my_agent", [stub_connection])
+    identity = Identity("my_agent")
```

Comment:
the identity needs to be constructed outside the agent.

Reply:
must it be the case, always?

Reply:
if the public keys and addresses must match with the `Wallet`'s ones, then it would be easier to initialize them in the `Agent`/`AEA` constructor.

Reply:
yes in my opinion, otherwise, as we enrich the identity concept we will break the Agent/AEA API.

Reply:
The public keys and addresses must not match the wallet ones in general

Reply:
How comes that the public keys and addresses must not match the wallet ones? I mean, I don't see the point having different pbk and addresses from the wallet.

Reply:
The question is why they should have to match. I could see a scenario where I want to use one address for interacting with AEAs in public and another to do payments with the subset of agents who I actually transact with. So forcing the two to align avoids this.

Reply:
Though both of the keys should be inside the wallet and must be a pair private/public key. Otherwise we need to find an other way to identify the agents. I mean, if the address does not divert from the public key there is no reason to be in the wallet. And also it must not be of type address :)